src/app/clusters/camera-av-stream-management-server/camera-av-stream-management-server.cpp

```diff
@@ -0,0 +1,1801 @@
+{
+    uint8_t buffer[kViewportStructMaxSerializedSize];
+    TLV::TLVWriter writer;
+
+    writer.Init(buffer);
+    ReturnErrorOnFailure(viewport.Encode(writer, TLV::AnonymousTag()));
+
+    return mPersistentStorage->SyncSetKeyValue(DefaultStorageKeyAllocator::CameraAVStreamMgmtViewport().KeyName(), buffer,
+                                               static_cast<uint16_t>(writer.GetLengthWritten()));
+}
+
+CHIP_ERROR CameraAVStreamMgmtServer::ClearViewport()
+{
+    // TODO
+    return CHIP_NO_ERROR;
+}
+
+CHIP_ERROR CameraAVStreamMgmtServer::LoadViewport(ViewportStruct & viewport)
+{
+    uint8_t buffer[kViewportStructMaxSerializedSize];
+    MutableByteSpan bufferSpan(buffer);
+    uint16_t size = static_cast<uint16_t>(bufferSpan.size());
+
+    ReturnErrorOnFailure(mPersistentStorage->SyncGetKeyValue(DefaultStorageKeyAllocator::CameraAVStreamMgmtViewport().KeyName(),
+                                                             bufferSpan.data(), size));
+    bufferSpan.reduce_size(size);
+
+    TLV::TLVReader reader;
+
+    reader.Init(bufferSpan);
+    ReturnErrorOnFailure(reader.Next(TLV::AnonymousTag()));
+    ReturnErrorOnFailure(viewport.Decode(reader));
+
+    return CHIP_NO_ERROR;
+}
+
+CHIP_ERROR CameraAVStreamMgmtServer::StoreRankedVideoStreamPriorities()
+{
+    uint8_t buffer[kRankedVideoStreamPrioritiesTlvSize];
+    TLV::TLVWriter writer;
+    writer.Init(buffer);
+
+    TLV::TLVType arrayType;
+    ReturnErrorOnFailure(writer.StartContainer(TLV::AnonymousTag(), TLV::kTLVType_Array, arrayType));
+
+    for (uint8_t i = 0; i < kNumOfStreamUsageTypes; i++)
+    {
+        ReturnErrorOnFailure(writer.Put(TLV::AnonymousTag(), mRankedVideoStreamPriorities[i]));
+    }
+    ReturnErrorOnFailure(writer.EndContainer(arrayType));
+
+    return mPersistentStorage->SyncSetKeyValue(
+        DefaultStorageKeyAllocator::CameraAVStreamMgmtRankedVideoStreamPriorities().KeyName(), buffer,
+        static_cast<uint16_t>(writer.GetLengthWritten()));
+}
+
+CHIP_ERROR CameraAVStreamMgmtServer::LoadRankedVideoStreamPriorities()
+{
+    uint8_t buffer[kRankedVideoStreamPrioritiesTlvSize];
+
+    uint16_t len = kRankedVideoStreamPrioritiesTlvSize;
+    ReturnErrorOnFailure(mPersistentStorage->SyncGetKeyValue(
+        DefaultStorageKeyAllocator::CameraAVStreamMgmtRankedVideoStreamPriorities().KeyName(), buffer, len));
+
+    TLV::TLVReader reader;
+    reader.Init(buffer);
+
+    ReturnErrorOnFailure(reader.Next(TLV::kTLVType_Array, TLV::AnonymousTag()));
+    TLV::TLVType arrayType;
+    ReturnErrorOnFailure(reader.EnterContainer(arrayType));
+
+    for (uint8_t i = 0; i < kNumOfStreamUsageTypes; i++)
+    {
+        if (reader.Next(TLV::kTLVType_UnsignedInteger, TLV::AnonymousTag()) != CHIP_NO_ERROR)
+        {
+            break;
+        }
+
+        ReturnErrorOnFailure(reader.Get(mRankedVideoStreamPriorities[i]));
+    }
+
+    ReturnErrorOnFailure(reader.ExitContainer(arrayType));
```

Comment:
Whoa, wait.  If our Next() failed because the TLV was malformed, we need to return at that point.  Next() is only allowed to return "end of TLV" or success for us to keep going.